apps/noodle/librapi2/trunk/tools/man/prm.1
.\" $Id: prm.1 2348 2006-04-05 14:58:44Z twogood $
.TH "PRM" "1" "November 2002" "The SynCE project" "http://synce.sourceforge.net/"
.SH NAME
prm \- remove file

.SH SYNOPSIS
\fBprm\fR [-d \fILEVEL\fR] [-p \fIDEVPATH\fR] [-h] \fIFILE\fR

.SH "DESCRIPTION"

.PP
\fBprm\fR removes a file on a device connected through SynCE.

.PP
Forward slashes ('/') on the command line are converted to backward slashes ('\\').

.SH "OPTIONS"

.TP
-d \fILEVEL\fR
Set debug log level:
.IP
0 - No logging (default)
.IP
1 - Errors only
.IP
2 - Errors and warnings
.IP
3 - Everything

.TP
-p \fIDEVPATH\fR
Use the device at the given path, instead of the default.

.TP
-h
Display help message.

.TP
\fIFILE\fR
Full path name to the file that should be removed.

.SH "AUTHOR"
.PP
This manual page was written by David Eriksson <twogood@users.sourceforge.net>.
.SH "SEE ALSO"
synce(1) pcp(1) pls(1) pmv(1) pmkdir(1) prmdir(1)


Changelog
BETA3 (n/a)
===========

PERSONALIZACIÓN Y CONFIGURACIÓN DEL SISTEMA (BRANDING)
------------------------------------------------------

- Cerrada lista prácticamente definitiva de aplicaciones
- Incluidas nuevas aplicaciones:
  - lsr
  - mozilla-acroread
  - dasher
  - gFTP (BTS 1087)
  - xchat (BTS 1086)
  - bluefish (BTS 1085)
  - d4x (BTS 1084)
  - soundtracker (BTS 1063)
  - gnome-games
  - firestarter
  - streamtuner
  - aMSN (v. 0.96)
  - gdebi
  - audacity (BTS 1059)
- Incluidas aplicaciones de consola (del spec): mc, ncftp, minicom, vim, ssh,
wget, rsync, dillo, sqlite, nmap, gsm-utils, brltty, joe
- Mejoras en el sistema de suplementos: comprobación de si el suplemento se
instala correctamente, comprobación de dependencias,...
- Adaptación de "contenido para Windows" de la v3 a la v4: pendiente de
terminar de depurar (BTS 1044)
- Solucionado problema de acceso a unidades SAMBA de la red: no se refrescaban
correctamente las unidades de red a través de nautilus
- Incluidos ficheros de ejemplo, adaptando los de la v3
- Pruebas de hibernación y gestión de energía en equipos portátiles correctas
- Añadido asistente para instalación de codecs de audio/vídeo
- Homogeneizadas entradas del menú: descripción (nombre aplicación)
- Implementado e incluido asistente para conexión a Internet
- Incluido "watermain"
- Traducido "gxine" (al final, descartada de la distribución base)
- Incluida imagen de fondo de Grub
- Personalizada página de prueba de impresión
- Configuración adecuada de "rhythmbox": inclusión de algunas radios
nacionales (tal y como se indica en el "spec")
- Configuración adecuada de "liferea": inclusión de algunos periódicos
nacionales (tal y como se indica en el "spec")
- Las aplicaciones en "Herramientas de Sistema" pasan a "Sistema ->
Administración"
- Configuración adecuada de "amule": navegador y reproductor de vídeo por
defecto (BTS 1017)


INSTALADOR LIVE (LIVE-INSTALLER)
--------------------------------

- Añadido visualizador de particiones de disco en selección de puntos de
montaje
- Mejorado (más usable y sencillo) el resumen final del instalador (BTS 955)
- Corregido problema de instalación de paquetes de soporte de inglés al final
de la instalación (ya no los instala)
- Corregido error de instalación cuando existen múltiples discos duros (BTS
965 y 1126)
- Mejorado sistema de redimensionado: permitía disminuir el tamaño por debajo
del mínimo necesario para la instalación
- Mejorado opción de utilizar espacio libre: permitía usar espacio sin
particionar demasiado pequeño
- Añadido soporte para instalación del perfil de accesibilidad con el que se
arranca en live al usuario que se crea en el sistema instalado
- Mejoradas descripciones de opciones de particionado
- Corregidos errores de unidades montadas tras usar "DiskPreview"
- Establecidos valores mínimos de instalación: 3 GB (/), 500 MB (/home), 64 MB
(swap)
- Análisis de soporte de accesibilidad de "gparted": habría que hacer un "fork"
- Personalizado botón de instalar incluyendo nuevo logo

INSTALADOR NO LIVE (DI-INSTALLER)
---------------------------------

- Corregidos algunos problemas leves de traducción

ACCESIBILIDAD
-------------

- Análisis y repaso de perfiles de accesibilidad
- Validación de funcionamiento de los perfiles de accesibilidad

NOODLE
------

- Implementado e incluido actor de Hermes para gestión de PDAs
- Finalizada integración en Guadalinex: paquetes guadalinex-noodle
- Análisis de la tecnología "opensync" para sincronización a través de
"syncml" (eg: symbian); probadas diferentes versiones (0.18 (dapper), 0.19,
020svn): descartado por considerarse poco estable aún.
- Corregidos algunos errores.

SISTEMA DE GENERACIÓN (GENERATION SYSTEM)
-----------------------------------------

- Actualizados paquetes relevantes de "-updates", "-security" y "-backports"
- Mejorada la estructura de metapaquetes de Guadalinex para hacer más flexible
la desinstalación de paquetes sin afectar a las configuraciones del resto

E-ADMIN
-------

- Incluido sistema de instalación de certificado raíz de la FNMT:
  - Cuando se detecta un certificado de usuario y se va a instalar en Firefox,
  se pide permiso al usuario para la instalación del certificado raíz,
  pidiendo que acepte la licencia.
  - Hecho lo anterior, se instala el certificado raíz en el sistema
- Modificado sistema de creación de usuarios para añadir a los nuevos usuarios
al grupo "scard"
- Añadidas librerías "jss" para acceso a diferentes servicios Java de la JA
- Análisis de firma en Evolution con DNIe: descartado (no es posible firmar
con el DNIe)


BETA2 (12/01/07)
================

PERSONALICACIÓN Y CONFIGURACIÓN DEL SISTEMA (BRANDING)
------------------------------------------------------

- Incluido "ndiswrapper" para soporte de algunas tarjetas wifi
- Añadidos iconos en el escritorio: "carpeta personal" y "equipo".
- Repaso e inclusión en Guadalinex de todos los paquetes interesantes que
incluye Ubuntu Edgy y que no estaban en Guadalinex: evolution-exchange,...
- Añadidos repositorios oficiales de Guadalinex v4
- Implementado interfaz de usuario para suplementos de manera que se pueda
utilizar sin hermes
- Compartición del directorio "Compartido" para permitir a los usuarios
compartir ficheros a través de SAMBA
- Regenerados "grub" y "grub-installer" para configurar adecuadamente el
nombre de la distribución ("Ubuntu" -> "Guadalinex")
- Comprobación de detección de entradas de GRUB para otros sistemas
operativos tras instalación
- Implementación de nueva funcionalidad para gnome-system-tools: soporte de
perfiles de accesibilidad para los usuarios que se crean
- Establecido número de escritorios virtuales a 4
- Añadido "alacarte" para edición de menús
- Añadido "camorama" para uso de webcams
- Añadido "lemurae" para acceso al diccionario de la RAE
- Traducido mensaje inicial al abrir una consola
- Personalizado el icono de salida de sesión
- Deshabilitado servidor SSH por defecto

apps/noodle/librapi2/trunk/tools/prm.c
/* $Id: prm.c 2348 2006-04-05 14:58:44Z twogood $ */
#include "pcommon.h"
#include <rapi.h>
#include <synce_log.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

char* devpath = NULL;

static void show_usage(const char* name)
{
	fprintf(stderr,
			"Syntax:\n"
			"\n"
			"\t%s [-d LEVEL] [-p DEVPATH] [-h] FILE\n"
			"\n"
			"\t-d LEVEL     Set debug log level\n"
			"\t                 0 - No logging (default)\n"
			"\t                 1 - Errors only\n"
			"\t                 2 - Errors and warnings\n"
			"\t                 3 - Everything\n"
			"\t-h           Show this help message\n"
                        "\t-p DEVPATH   Device path\n"
			"\tFILE         The remote file you want to remove\n",
			name);
}

static bool handle_parameters(int argc, char** argv, char** path)
{
	int c;
	int log_level = SYNCE_LOG_LEVEL_LOWEST;

	while ((c = getopt(argc, argv, "d:p:h")) != -1)
	{
		switch (c)
		{
			case 'd':
				log_level = atoi(optarg);
				break;
			
                        case 'p':
                                devpath = optarg;
                                break;
			
			case 'h':
			default:
				show_usage(argv[0]);
				return false;
		}
	}

	synce_log_set_level(log_level);

	if (optind == argc)
	{
		fprintf(stderr, "%s: No file specified on command line\n\n", argv[0]);
		show_usage(argv[0]);
		return false;
	}
		
	/* TODO: handle more than one path */
	*path = strdup(argv[optind]);

	return true;
}

int main(int argc, char** argv)
{
	int result = 1;
        char* path = NULL;
        RapiConnection* connection = NULL;
	HRESULT hr;
	WCHAR* wide_path = NULL;
	
	if (!handle_parameters(argc, argv, &path))
		goto exit;

        if ((connection = rapi_connection_from_path(devpath)) == NULL)
        {
          fprintf(stderr, "%s: Could not find configuration at path '%s'\n", 
                  argv[0],
                  devpath?devpath:"(Default)");
          goto exit;
        }
        rapi_connection_select(connection);
	hr = CeRapiInit();

	if (FAILED(hr))
	{
		fprintf(stderr, "%s: Unable to initialize RAPI: %s\n", 
				argv[0],
				synce_strerror(hr));
		goto exit;
	}


	convert_to_backward_slashes(path);
  
  if (':' == path[0])
	  wide_path = wstr_from_current(path + 1);
  else
	  wide_path = wstr_from_current(path);
  
	wide_path = adjust_remote_path(wide_path, true);

	if (!CeDeleteFile(wide_path))
	{
		fprintf(stderr, "%s: Failed to remove '%s': %s\n", 
				argv[0],
				path,
				synce_strerror(CeGetLastError()));
		goto exit;
	}

	result = 0;

exit:
	if (wide_path)
		wstr_free_string(wide_path);

	if (path)
		free(path);

	CeRapiUninit();
	return result;
}
